src/test/java/com/microsoft/sqlserver/jdbc/SQLServerConnectionTest.java

```diff
@@ -934,6 +935,58 @@ public void run() {
         assertTrue(status && future.isCancelled(), TestResource.getResource("R_threadInterruptNotSet"));
     }
 
+
+   /**
+    * Test thread count when finding socket using threading.
+    *
+    * @throws InterruptedException
+    *         If any thread has interrupted the current thread.
+    */
+   @Test
+   @Tag(Constants.xAzureSQLDB)
+   @Tag(Constants.xAzureSQLDW)
+   public void testThreadCountWhenFindingSocket() throws InterruptedException {
+       ExecutorService executor = null;
+       ManagementFactory.getThreadMXBean().resetPeakThreadCount();
+
+       try {
+           executor = Executors.newSingleThreadExecutor(r -> new Thread(r, ""));
+           executor.submit(() -> {
+               try {
+                   SQLServerDataSource ds = new SQLServerDataSource();
+                   ds.setServerName("localhost");

```

Comment:
we shouldn't assume there's a local server running, test should not fail if no local server